flytekit/core/data_persistence.py

```diff
@@ -108,6 +110,29 @@ def get_fsspec_storage_options(
     return {}
 
 
+def get_additional_fsspec_call_kwargs(protocol: typing.Union[str, tuple], method_name: str) -> Dict[str, Any]:
+    """
+    These are different from the setup args functions defined above. Those kwargs are applied when asking fsspec
+    to create the filesystem. These kwargs returned here are for when the filesystem's methods are invoked.
+
+    :param protocol: s3, gcs, etc.
+    :param method_name: Pass in the __name__ of the fsspec.filesystem function. _'s will be ignored.
+    """
+    kwargs = {}
+    method_name = method_name.replace("_", "")
+    if isinstance(protocol, tuple):
+        protocol = protocol[0]
+
+    if protocol == "s3":
+        if method_name == "put":
+            kwargs["chunksize"] = _WRITE_SIZE_CHUNK_BYTES
+    if protocol == "gs":
+        if method_name == "put":
+            kwargs["chunksize"] = _WRITE_SIZE_CHUNK_BYTES
```

Comment:
what do you think of 
```suggestion
    # TODO: Comment explaining the issue
    if method_name == "put" and protocol in ["s3", "gs"]:
            kwargs["chunksize"] = _WRITE_SIZE_CHUNK_BYTES
```